rfc/rfc-0000-move-off-monorepo.md

```diff
@@ -0,0 +1,57 @@
+- extra support burden on the current maintainers
+- maintaining development and style consistency
+- existing CI, release automation and GitHub tooling doesn't work well with monorepos
+- as we continue to build up the Cartographer ecosystem with integrations, adaptors, platform controllers this will cause a bottleneck for changes proposed on a single repo.
+
+Adopting a microservice approach for Cartographer will aid extensions and help build up the wider ecosystem and community around it.
+This RFC proposes how we will adopt this microservice style approach that will help grow the Cartographer ecosystem, encourage new integrations and extensions is a sustainable and maintainable way that suits the project and its aspirations.
+
+## Motivation
+
+The motivation behind Cartographer as an OSS project aims to build a community and ecosystem to aid supply chain choreography.  Cartographer itself is an "umbrella" for this ecosystem, the first step of this was to build the core components and now we are looking ahead to extend by adding more capabilities.
+ 
+As expressed by maintainers of the https://github.com/vmware-tanzu/cartographer repository it will become hard to manage it if we use it as a monorepo and continue to add more capabilities like conventions etc.
+It is expected that this RFC describes how we split the current repository so it allows the Cartographer OSS project to evolve and scale beyond the concrete capabilities the core aims to provide.
+For example splitting out from the current repo:
+
+-  the __`Runnable`__ feature so it can release independently and install standalone in a target deployment cluster,  avoids deploying all of Cartographer which contains build cluster CRDs not desired in runtime clusters
+- the __examples directory__ so we can build up a wide range of examples that include integrations with wider platform services like conventions.  This also helps encourage contribution of community supply chains for different integrations where we can add automated testing that runs against different setups.
+- the __website__ so we can add documentation for these new integrations and platform capabilities outside of the core repository.  I.e. Documenting services like the conventions or a Jenkins integration should not require a change to the core cartographer repository, this should live outside in its own repository.
+
+## Possible Solutions
+
+There seems to be only two options possible solutions:
+- __monorepo__ we could revisit the [convention proposal](https://github.com/vmware-tanzu/cartographer/pull/514) decision and look at ways to address the monorepo concerns however there seemed a general consensus on the [office hours call](https://www.youtube.com/watch?v=iCURQqV52Uw&t=1173s) that this was not desired.
+- __microservice__ As conventions are now being worked on in a different repository, we should look at splitting out parts that currently reside in https://github.com/vmware-tanzu/cartographer.
+
+Suggestions:
```

Comment:
Would it also make sense to split ClusterSupplyChain/Workload from ClusterDelivery/Deliverable? These components are highly likely to be installed into distinct clusters.

Reply:
> These components are highly likely to be installed into distinct clusters.

That is true. Do we have a reasonable expectation that users would object to installing both CRDs in a cluster only to leverage one? Would the value delivered match the work of separating the code? (As they do share code)

Reply:
Please see https://github.com/vmware-tanzu/cartographer/pull/571#discussion_r796704119 wrt priorities for this work. I think ultimately it would be nice, but until we see driving factors as @waciumawanjohi suggests, it's not a _need_.

Reply:
I've certainly seen pushback from end users for installing build related components (including CRDs) into a target run cluster.  Having a standalone / microservice installation for this makes sense to me.